### PR TITLE
ZCS-14744 : Update license activation validation API

### PR DESCRIPTION
**Pull request by umagmrit**
While upgrading Zimbra there is an activation check executed to check whether the license is valid to use this version, presently upgrade script makes this call to the following API with some GET params.

`https://$host/zimbraLicensePortal/public/activation?action=getActivation&licenseId=$licenseId&version=$options{version}&fingerprint=$fingerprint&activationId=$activationId`
But for the real-time licensing, there is another API that will be used instead

Introducing a new option, `--licenseKey`, which can be passed to the upgrade or install script (./install.sh). This ensures that the licenseCheck is executed for the upgrade component.

Example Usage:
/install.sh --licenseKey xxxxxx4455xx455xxxxx

If the administrator fails to provide the license key, a prompt should be issued during installation.

Deleting below files before upgrade as these are not supported with new license changes.
```
/opt/zimbra/conf/ZCSLicense.xml
/opt/zimbra/conf/ZCSLicense-Trial.xml
/opt/zimbra/conf/ZCSLicense-activated.xml
```

### CODE REVIEW

File: rpmconf/Build/checkLicense.pl [hunk 1]
```diff
@@ -94,40 +90,37 @@ sub usage {
...
+my $response = $browser->request($request);
 if ( $response->is_success ) {
-    myDie(0,"SUCCESS: ", $response->content, "\n");
-}
-else {
-    myDie(1,"ERROR: ", $response->content, "\n");
+	my $json_content = $response->content;
+	my ($status_code) = $json_content =~ /"status":\s*(\d+)/;
+	if (defined $status_code && $status_code == 2000) {
+		myDie(0,"SUCCESS: ", $response->content, "\n");
```
Comment (dasiyogesh) [inline rpmconf/Build/checkLicense.pl]:
```suggestion
		myDie(0, "SUCCESS: ", $response->content, "\n");
```

File: rpmconf/Build/checkLicense.pl [hunk 2]
```diff
@@ -94,40 +90,37 @@ sub usage {
...
+	} else {
+		myDie(1,"ERROR: ", $response->content, "\n");
```
Comment (dasiyogesh) [inline rpmconf/Build/checkLicense.pl]:
```suggestion
		myDie(1, "ERROR: ", $response->content, "\n");
```

File: rpmconf/Build/checkLicense.pl [hunk 3]
```diff
@@ -94,40 +90,37 @@ sub usage {
...
+	}
+} else {
+	myDie(1,"ERROR: ", $response->content, "\n");
```
Comment (dasiyogesh) [inline rpmconf/Build/checkLicense.pl]:
```suggestion
	myDie(1, "ERROR: ", $response->content, "\n");
```

**Review by RakeshAMore — APPROVED** [review]

**Review by chitranshu198 — APPROVED** [review]

File: rpmconf/Build/checkLicense.pl [hunk 4]
```diff
@@ -82,52 +111,57 @@ sub usage {
...
 my @lwpargs = -f $caf ? ( ssl_opts => { SSL_ca_file => $caf, SSL_ca_path => undef } ) : ();
 my $browser = LWP::UserAgent->new(@lwpargs);
 $browser->env_proxy;
+my $request = HTTP::Request->new(POST => "$host/rest/v1/public/license/$licensekey/validate?version=$options{upgradeVersion}");
+$request->header('Content-Type' => 'application/json');
+my $response = $browser->request($request);
+if ( $response->is_success ) {
+	my $json_content = $response->content;
+	my ($status_code) = $json_content =~ /"status":\s*(\d+)/;
+	if (defined $status_code && $status_code == 2000) {
```
Comment (dasiyogesh) [inline rpmconf/Build/checkLicense.pl:156]: this should be 200 right?

Comment (dasiyogesh) [reply rpmconf/Build/checkLicense.pl:156]: its fine, this is 2000 as per the api.

**Review by dasiyogesh — APPROVED** [review]

**Review by RakeshAMore — APPROVED** [review]

**Pull request merged**